spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -88,6 +88,13 @@ sealed class SpdxExpression {
         }
     }
 
+    /**
+     * Return all single licenses contained in the expression tree whereas a single license is one of:
+     * [SpdxLicenseIdExpression], [SpdxLicenseReferenceExpression] or [SpdxCompoundExpression] using [SpdxOperator.WITH]
+     * as operator.
+     */
+    abstract fun decompose(): List<SpdxExpression>
```

Comment:
As the order is not important (and you call `distinct` anyway), how about making the return type an `EnumSet<SpdxLicense>` like for the current `spdxLicenses()`?

And as a side note, once this PR is merged I guess we should spend some little more clean-up work to get completely rid of the `licenses()` and `spdxLicenses()` functions, as I like the new `decompose()` functionality much better, and to avoid confusion when to use which function.

Reply:
I'm using a `Set<SpdxExpression>` now as using that `SpdxLicense` does not quite fit since e.g. also `License-Ref-*` licenses are involved.

Regarding the `clean-up`, I had that in mind as well. Will do a PR later on.